### PR TITLE
cd3-boost-unit-definition updates

### PR DESCRIPTION
**Pull request by CD3**
Some additional cleanup and changes from the last review.

Changed the boost dependency to a range [>=1.69.0]

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/cd3-boost-unit-definitions/all/conanfile.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def layout(self):
         basic_layout(self)
 
     def requirements(self):
-        self.requires("boost/1.72.0", transitive_headers=True)
+        self.requires("boost/[>=1.69.0]", transitive_headers=True)
```
Comment (uilianries) [inline recipes/cd3-boost-unit-definitions/all/conanfile.py:44]: Ranged version is not allowed in CCI, use only specific version please.

Comment (CD3) [reply recipes/cd3-boost-unit-definitions/all/conanfile.py:44]: Oops, sorry. So what is the recommendation for selecting the version? Should it be the newest version available at the time, or the oldest version supported?

Comment (prince-chrismc) [reply recipes/cd3-boost-unit-definitions/all/conanfile.py:44]: Newest available :)

Comment (prince-chrismc) [reply recipes/cd3-boost-unit-definitions/all/conanfile.py:44]: Todo for me but the docs are sucky on this one

https://github.com/conan-io/conan-center-index/blob/3b5742afddd89408c6343dd18b173283fbb84818/docs/faqs.md?plain=1#L355

https://github.com/search?q=repo%3Aconan-io%2Fconan-center-index%20version%20ranges&type=code

Comment (CD3) [reply recipes/cd3-boost-unit-definitions/all/conanfile.py]: You got it.

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**